### PR TITLE
Improve Validation of Config

### PR DESCRIPTION
**Pull request by HTRamsey**
Improve Validation of Config

Description
-----------
Verifies FreeRTOSConfig.h values that are required
Uses static asserts on a few ipconfig values that can't be compile time checked normally

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by paulbartell** [comment]
@HTRamsey Thanks for the PR. Would you mind running uncrustify on your changes?

**Review by paulbartell — DISMISSED** [review]

File: source/include/FreeRTOSIPConfigDefaults.h [hunk 1]
```diff
@@ -136,19 +136,51 @@
...
+    #error INCLUDE_vTaskDelay must be set to 1
+#endif
+
+#if ( INCLUDE_xTaskGetCurrentTaskHandle == 0 )
+    #error INCLUDE_xTaskGetCurrentTaskHandle must be set to 1
+#endif
+
+#if ( configSUPPORT_DYNAMIC_ALLOCATION == 0 )
+    #error configSUPPORT_DYNAMIC_ALLOCATION must be set to 1
+#endif
```
Comment (AniruddhaKanhere) [inline source/include/FreeRTOSIPConfigDefaults.h:174]: Yes, this is good. We can remove this once https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/issues/449 has been resolved.

**Review by AniruddhaKanhere — APPROVED** [review]

**Comment by HTRamsey** [comment]
@tony-josi-aws we can change the static assert definition, that semi colon is just to allow multiple asserts on the same line.

**Comment by tony-josi-aws** [comment]
@HTRamsey, makes sense to remove the semicolon if we are not using multiple asserts on the same line.

And MISRA doesn't like literals without a sign (U) explicitly specified.

**Comment by HTRamsey** [comment]
@tony-josi-aws Should I add literal signs to all of the default configs?

**Review by tony-josi-aws — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Comment by HTRamsey** [comment]
Is there an actual minimum kernel version that is supported?

**Pull request merged**